chore(deps): Update c-toxcore to v0.2.18

anthonybilinski opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6625)
<!-- Reviewable:end -->

anthonybilinski committed:
chore(deps): Update c-toxcore to v0.2.18

Replace msgpack-c with toxcore submodule of cmp.

sudden6 commented:
c-toxcore recently introduced a git submodule, which probably needs to be checked out additionally for builds to pass again.

robinlinden commented:
msgpack-c was also dropped, so unless you're using that outside of c-toxcore, it can be removed.

anthonybilinski commented:
We download the source bundle rather than git cloning, so the submodule can't be checked out directly, but I've changed the download archive to the newly added source bundle: https://github.com/TokTok/c-toxcore/issues/2192.

Also removed all msgpack-c scripts.
